src/App.tsx

```diff
@@ -1,31 +1,79 @@
 import './styles/App.scss';
-import ResultsSection from './components/ResultsSection/ResultsSection';
+import CardList from './components/CardList/CardList';
 import SearchSection from './components/SearchSection/SearchSection';
-import { AppState, Movie } from './types';
-import { Component } from 'react';
+import { Movie } from './types';
+import { useCallback, useEffect, useState } from 'react';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
+import { Paths } from './router/routesConstants';
+import Pagination from './components/Pagination/Pagination';
 
-class App extends Component<object, AppState> {
-  state: AppState = {
-    results: [],
-    loading: false,
+export default function App() {
+  const [results, setResults] = useState<Movie[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [totalPages, setTotalPages] = useState<number>(0);
+  const location = useLocation();
+  const navigate = useNavigate();
+  const isDetailsOpen = location.pathname === Paths.DETAILS;
```

Comment:
You could use useMatch hook for this. It's not necessary, but a little more neat.

https://reactrouter.com/en/main/hooks/use-match

Reply:
Thanks for advice.